Source/Scene/BillboardCollection.js

```diff
@@ -255,6 +266,16 @@ define([
                 return that._textureAtlas.texture;
             }
         };
+
+        this._removeEventFunc = undefined;
+        if (defined(this._globe)) {
+            this._removeEventFunc = this._globe._surface.tileRenderedEvent.addEventListener(function(tile) {
+                var tileList = that._renderedTileList;
+                if (tileList.indexOf(tile) === -1) {
```

Comment:
The linear search happens whenever a tile becomes visible. The only time it matters is the case you gave. This was an issue at one point when the billboard height was changing to match the tile. Now that we keep the height of the highest res tile, it shouldn't be a problem anymore.